coalib/results/Diff.py

```diff
@@ -70,6 +72,91 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unified_diff(cls, unified_diff, original_file):
+        """
+        Creates a ``Diff`` object from given unified diff.
+
+        If the provided unified diff does not contain any patch,
+        the ``Diff`` object initialized from the original file is
+        returned.
+
+        :param unified_diff:  Unified diff string.
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        :raises RuntimeError: Raised when the context lines or the
+                              lines to be removed do not match in
+                              the original file and the unified diff.
+        """
+        patch_set = PatchSet(unified_diff.splitlines())
+
+        diff = Diff(original_file)
+
+        if not patch_set:
+            return diff
+
+        file = patch_set[0]
+
+        for hunk in file:
+            file_line = hunk.source_start
+            hunk_iterator = iter(hunk)
+
+            try:
+                while True:
+                    line = next(hunk_iterator)
+                    source_code = str(line)[1:]
+                    if line.is_added:
+                        add_set = []
+                        # As ``Diff`` does not allow line additions to a
+                        # position more than one time, add all the
+                        # consecutive '+' lines at once.
+
+                        try:
+                            while line.is_added:
+                                source_code = str(line)[1:]
```

Comment:
I think you can move this directly below `line = next(hunk_iterator)`, because at this stage `source_code` does already point to the correct line (see L106).
-->
```python
while line.is_added:
    add_set.append(source_code)
    line = next(hunk_iterator)
    source_code = str(line)[1:]

diff.add_lines(file_line-1, add_set)
#source_code = str(line[1:]) <-- can be removed now
```